src/WalletInit/ConnectNanoX/ConnectNanoXScreen.tsx

```diff
@@ -51,9 +51,9 @@ export const ConnectNanoXScreen = ({defaultDevices}: Props) => {
     }
   }
 
-  const onConnectBLE = (deviceId: DeviceId) => {
+  const onConnectBLE = async (deviceId: DeviceId) => {
     Logger.debug('deviceId', deviceId)
-    getHWDeviceInfo(walletImplementationId, deviceId, null, useUSB).then(onSuccess).catch(onError)
+    await getHWDeviceInfo(walletImplementationId, deviceId, null, useUSB).then(onSuccess).catch(onError)
```

Comment:
This had to be `async` and `await` as we expect this function to be pending for a few moments in the `_onSelectDevice` method. This caused the app to be laggy and unresponsive.